impeller/renderer/backend/vulkan/debug_context_vk.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include "flutter/fml/macros.h"
+#include "impeller/renderer/backend/vulkan/vk.h"
+
+namespace impeller {
+
+class DebugContextVK {
+ public:
+  explicit DebugContextVK(const vk::Device& logical_device);
+
+  ~DebugContextVK();
+
+  template <typename T>
+  bool SetDebugName(T handle, std::string_view label) const {
+    uint64_t handle_ptr =
+        reinterpret_cast<uint64_t>(static_cast<typename T::NativeType>(handle));
+
+    std::string label_str = std::string(label);
+
+    auto ret = logical_device_.setDebugUtilsObjectNameEXT(
```

Comment:
In the GLES backend, debug object labelling is [handled by the reactor](https://github.com/flutter/engine/blob/9b278ae58fbe9e94a174ea3dd493960103f5a400/impeller/renderer/backend/gles/reactor_gles.h#L51). Perhaps we can do something similar in the Vulkan backend and move this call to the `ContextVK`. A context is always available and it always has a device handle ready.

IMO that would be a lot simpler and keeping in line with the other API backends.

Reply:
This is resolved as well.